kessler/cdm.py

```diff
@@ -191,11 +191,16 @@ def __eq__(self, other):
         if isinstance(other, ConjunctionDataMessage):
             return hash(self) == hash(other)
         return False
-
+    
     def set_header(self, key, value):
         if key in self._keys_header:
             if key in self._keys_with_dates:
                 # We have a field with a date string as the value. Check if the string is in the format needed by the CCSDS 508.0-B-1 standard
+                timeFormat = time_utils.getCcsdsTimeFormat(value)
+                idx = timeFormat.find('DDD')
+                if idx!=-1:
+                    [DateNum, DateVec] = time_utils.DOY2Date(value[idx:idx+3], value[0:4])
+                    value = str(DateVec[0]) +'-' + str(DateVec[1]) + '-' + str(DateVec[2]) + 'T' + value[idx+4:-1] #figure this out!!!
```

Comment:
I think that you could simplify this by only returning the `datetime` object from the function `time_utils.doy_to_date`, and extract the year, day and month in line 203, directly from the `datetime` object

Reply:
Agree with this.